metrics/resource_view_test.go

```diff
@@ -341,19 +343,32 @@ testComponent_testing_value{project="p1",revision="r2"} 1
 			UnregisterResourceView(gaugeView, resourceCounter)
 
 			records := []metricExtract{}
-			for record := range ocFake.published {
-				for _, m := range record.Metrics {
-					if len(m.Timeseries) > 0 {
-						labels := map[string]string{}
-						if record.Resource != nil {
-							labels = record.Resource.Labels
+		loop:
+			for {
+				select {
+				case record := <-ocFake.published:
+					if record == nil {
+						continue loop
+					}
+					for _, m := range record.Metrics {
+						if len(m.Timeseries) > 0 {
+							labels := map[string]string{}
+							if record.Resource != nil {
+								labels = record.Resource.Labels
+							}
+							records = append(records, metricExtract{
+								Name:   m.MetricDescriptor.Name,
+								Labels: labels,
+								Value:  m.Timeseries[0].Points[0].GetInt64Value(),
+							})
 						}
-						records = append(records, metricExtract{
-							Name:   m.MetricDescriptor.Name,
-							Labels: labels,
-							Value:  m.Timeseries[0].Points[0].GetInt64Value(),
-						})
 					}
+					if len(records) >= len(expected) {
+						break loop
+					}
+				case <-time.After(4 * time.Second):
+					t.Error("Timeout reading input")
```

Comment:
I wonder if fatal is more appropriate here?

Reply:
Done.